src/Network/WebSockets/TSSocketManager.m

```diff
@@ -109,40 +109,132 @@ - (void)ensureWebsocket
     }
 
     // Try to reuse the existing socket (if any) if it is in a valid state.
-    SRWebSocket *socket = self.websocket;
-    if (socket) {
-        switch ([socket readyState]) {
+    if (self.websocket) {
+        switch ([self.websocket readyState]) {
             case SR_OPEN:
                 DDLogVerbose(@"WebSocket already open on connection request");
-                OWSAssert(self.status == kSocketStatusOpen);
```

Comment:
We can't assume socket and class state are aligned in `ensureWebsocket` - indeed, we need to do the work to ensure they are aligned.